### PR TITLE
Add question picker tests

### PR DESCRIPTION
**Pull request by WiNloSt**
This adds tests to `QuestionPicker` or when we're adding cards to a dashboard.

image

#### Testing plan for milestone 1
Dashboards
1. Add a question to a dashboard in a public collection
    **expectation**:
    - hides all personal collections
    - show all questions
1. Add a question to a dashboard in a personal collection
    **expectation**:
    - shows all collections
    - show all questions

### CODE REVIEW

**Comment by WiNloSt** [comment]
Current dependencies on/for this PR:
* master
  * **PR #35135** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35135" target="_blank">Graphite</a> 
    * **PR #35136** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35136" target="_blank">Graphite</a> 
      * **PR #35137** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35137" target="_blank">Graphite</a> 
        * **PR #35138** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35138" target="_blank">Graphite</a> 
          * **PR #35141** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35141" target="_blank">Graphite</a> 
            * **PR #35142** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35142" target="_blank">Graphite</a>  👈
          * **PR #35139** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35139" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/35142?utm_source=stack-comment).

File: frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+describe("AddCardSideBar", () => {
+  it("should render no items", async () => {
+    await setup({
+      collections: [],
+    });
+
+    expect(screen.getByText("Nothing here")).toBeInTheDocument();
+  });
+
+  it("should hide all personal collections when adding questions to dashboards in the root collection (public collection)", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx:121]: test Dashboards > #1 from [the epic](https://github.com/metabase/metabase/issues/34733)

File: frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    collectionItems.forEach(collectionItem => {
+      expect(
+        screen.getByRole("menuitem", {
+          name: collectionItem.name,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should show all questions when adding questions to dashboards in public collections", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx:179]: test Dashboards > #1 from [the epic](https://github.com/metabase/metabase/issues/34733)

File: frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    collectionItems.forEach(collectionItem => {
+      expect(
+        screen.getByRole("menuitem", {
+          name: collectionItem.name,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should show all collections when adding questions to dashboards in personal subcollections", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx:213]: test Dashboards > #2 from [the epic](https://github.com/metabase/metabase/issues/34733)

File: frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    assertBreadcrumbs([
+      ROOT_COLLECTION,
+      PERSONAL_COLLECTION,
+      PERSONAL_SUBCOLLECTION,
+    ]);
+
+    expect(screen.getByText("Nothing here")).toBeInTheDocument();
+  });
+
+  it("should show all questions when adding questions to dashboards in personal subcollections", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx:231]: test Dashboards > #2 from [the epic](https://github.com/metabase/metabase/issues/34733)

**Review by ranquild — APPROVED** [review]

**Pull request merged**